lib/lightning/ai_assistant/ai_assistant.ex

```diff
@@ -236,4 +265,27 @@ defmodule Lightning.AiAssistant do
        do: UsageLimiter.increment_ai_queries(session)
 
   defp maybe_increment_msgs_counter(_user_role), do: Multi.new()
+
+  @doc """
+  Updates the status of a specific message within a chat session.
+
+  Returns `{:ok, session}` if the update is successful, otherwise `{:error, changeset}`.
+  """
+  @spec update_message_status(ChatSession.t(), ChatMessage.t(), atom()) ::
+          {:ok, ChatSession.t()} | {:error, Changeset.t()}
+  def update_message_status(session, message, status) do
+    Multi.new()
+    |> Multi.update(
+      :message,
+      ChatMessage.changeset(message, %{status: status})
+    )
+    |> Multi.run(:session, fn _repo, _changes ->
+      {:ok, get_session!(session.id)}
+    end)
```

Comment:
I appreciate the concern with transactions. Can you simplify it a bit with `Repo.update`?